File: js/components/regex.js
// List of all the regex being used inside the linter.
//
// Extract each entry from the .bib file.
const entryPattern = /@(\w+)\{([^,]+),\s*(.*?)\},\s*\}/sg;
// Extract each field from the entry.
const fieldPattern = /(\w+)\s*=\s*(?:\{(.*?)\}|(\S+))/sg;

module.exports = { entryPattern, fieldPattern };

File: py/checks/components/regex.py
entry_pattern = re.compile(r'@(\w+)\{([^,]+),\s*(.*?)\},\s*\}', re.DOTALL)
field_pattern = re.compile(r'(\w+)\s*=\s*(?:\{(.*?)\}|(\S+))', re.DOTALL)